doc/compiler/tril/TrilLanguageReference.md

```diff
@@ -0,0 +1,194 @@
+# Tril Language Reference
+
+In may ways, the Tril language mirrors the Testarossa Intermediate Language,
+_Trees_. 
+
+This language reference intends to document mostly the places where special
+behaviour has been created in Tril to make writing test cases easier. 
+
+# Basics
+
+Tril is built out of two main components: 
+
+1. A very permissive S-expression parser.
+2. An IlGenerator that walks the AST produced by said parser, and generates
+   trees. 
+
+# Syntax
+
+Tril represents Trees using S-expressions, with key-value properties to augment
+a particular node with more information. 
+
+* `(x (y) (z))` represents some `x` with `y` and `z` as children.
+* `(x foo=bar baz=braz (y) (z)` is the same tree as above, but where the node
+  for `x` is augmented with two properties with values. 
+
+  Anything may be provided as a property for a node, however, unknown properties will be silently ignored.
```

Comment:
I think it would be useful to be more specific on the syntax requirements for literals and constants.  For example, can strings be quoted?  How do you specify floating point constants (10.0f)?  Can you use hex constants "0x1234" ?  etc.